### ISSUE TITLE
fckit: ICE when trying to build fckit with -g

### ISSUE THREAD

**Issue by pawosm-arm** [flang-compiler/flang#390]
flang2 suffers from ICE when trying to compile a module with **empty** derived typed variable when building it with -g.

I managed to reduce this issue to the following snippet that fails to build with -g:
```
module sometype_module

type sometype
end type sometype

type(sometype) :: something

end module sometype_module
```

The troublesome line is `type(sometype) :: something` and the error message is:
```
F90-S-0000-Internal compiler error. getsname: bad stype for       0  (sometype.f90: 8)
  0 inform,   0 warnings,   1 severes, 0 fatal for sometype_module
```

**Comment by pawosm-arm** [comment]
Could be false alarm, can't reproduce it using flang I've build 2nd of January.

**Comment by pawosm-arm** [comment]
Looks like some git merger went wrong.
